packages/common/amplify_db_common/lib/src/connect_io.dart

```diff
@@ -20,10 +22,11 @@ import 'package:path_provider/path_provider.dart';
 /// {@macro amplify_db_common.connect}
 QueryExecutor connect({
   required String name,
-  String? path,
+  FutureOr<String>? path,
 }) {
   return LazyDatabase(() async {
-    if (path == null) {
+    final resolvedPath = await path;
```

Comment:
```suggestion
    final resolvedPath = await (path ?? getApplicationSupportDirectory());
    return base.connect(name: name, path: resolvedPath);
```

Right now not passing a path here results in p.join being called twice (probably a miss on my part)

Reply:
Should be ?
```dart
    final resolvedPath =
        await path ?? (await getApplicationSupportDirectory()).path;
```

Reply:
Yeah